src/request.js

```diff
@@ -0,0 +1,43 @@
+// Updates the `Request` class to include the ability to manage the `cf`
+// features for the fetch requests.
+//
+// See https://developers.cloudflare.com/workers/reference/cloudflare-features/
+require('whatwg-fetch')
+
+// Available `cf` features. Any keys that are not here and set in `cf`
+// will be discarded.
+const cfFeatures = [
+  "scrapeShield",
+  "polish",
+  "minify",
+  "mirage",
+  "scrapeShield",
+  "apps",
+  "resolveOverride",
+  "cacheKey",
+  "cacheTtl",
+  "cacheTtlByStatus",
+
+  // Only available in SSL for SaaS product offering.
+  "hostMetadata"
+]
+
+const OldRequest = global.Request
+global.Request = class Request extends OldRequest {
+  constructor(input, options) {
+    super(input, options)
+
+    if (options && options.cf) {
+      let sanitisedOptions = {}
```

Comment:
Maybe this can be just `let sanitisedOptions = Object.keys(options.cf)....`? There seems no point create a empty object first.